Enable helloworld testcase for examples automation.

craigyam opened the pull request:
The helloworld tests are only run for examples automation
(make test.examples).  Re-enabling them to run for examples
using the k8s native routing rules.  The integration tests for
k8s will still be disabled until the bookinfo tests are updated.

craigyam committed:
Enable helloworld testcase for examples automation.

The helloworld tests are only run for examples automation
(make test.examples).  Re-enabling them to run for examples
using the k8s native routing rules.  The integration tests for
k8s will still be disabled until the bookinfo tests are updated.

elevran commented on testing/test-scripts/helloworld-v1-v2-route-rules.yaml:
```diff
@@ -24,3 +24,5 @@ spec:
         tags:
         - "version=v2"
+      - tags:
+        - "version=v1"
```
Don't understand why additional tag needed here.
Wanting to have v2 with 25% probability. Needs priority fix as in #512.

craigyam replied:
The script clears out the previous rules before the next tests.  Without setting version=v1 in that case 3/4 of the requests get a 404.  I can update the script to keep the default routing rule and then remove that.

elevran replied:
Or add the previous rule as well. Your call.

craigyam replied:
Done

elevran commented on testing/test-scripts/helloworld.sh:
```diff
@@ -51,26 +53,32 @@ MAX_LOOP=5
+    if [ "$ENV" == "docker" ]; then
+        create_rule $SCRIPTDIR/helloworld-default-route-rules.json
+    else
+        create_rule $SCRIPTDIR/helloworld-default-route-rules.yaml
```
At least currently we can create the json and yaml rules from the same source more or less. Something to consider so we don't have duplicate code?

elevran replied:
Also, seems like controlplane and demo topologies are copied to examples using a `docker` or `k8s` prefix, but rules are copied as is. In case there is ever a divergence, we should consider adding a prefix to the rules on copying as well.

craigyam replied:
The rules are currently not copied over to examples/ at all.  I'll update the automation to do that with the k8s prefix.

craigyam replied:
Will investigate generating rules files for automation.  For now, copy to examples/ and prefix appropriately.

elevran commented on testing/test-scripts/helloworld.sh:
```diff
@@ -82,9 +90,6 @@ while [  $retry_count -le $((MAX_LOOP)) ]; do
         echo ""
     fi
 
-    # Sleep a bit to make sure the rules are applied
```
Envoy still takes sometime to restart. Don't we need the sleep?

craigyam replied:
Sleep was added at 76 after rules creation.

elevran reviewed (commented):
Overall looks ok. Some clarifications needed per comments.

elevran reviewed (approved):
LGTM once comments are integrated into this PR

craigyam committed:
Address review comments.

- Copied the rules json/yaml to examples dir and prefix with
docker/k8s.
- Don't clear out rules after setting default route rule and
remove from v1-v2 route rule.
- Add copied rules json/yaml in examples to gitignore.
